app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java

```diff
@@ -130,7 +136,11 @@ public void verifyChannel(AuthenticationService.ChannelType channelType, Account
                 hibernateAccount.setStatus(AccountStatus.ENABLED);
             }
             hibernateAccount.setModifiedOn(DateUtils.getCurrentMillisFromEpoch());
-            hibernateHelper.update(hibernateAccount);
+            try {
+                hibernateHelper.update(hibernateAccount);    
+            } catch(PersistenceException pe) {
+                throw convertPersistenceException(pe, hibernateAccount.getId());
+            }
```

Comment:
I feel like in terms of mock tests, no. The question is open as to the full set of actual errors that could be thrown here. It appears in all cases, these will be wrapped in a PersistenceException, so I could add some tests of other exceptions that are not currently specifically accounted for, to be sure an acceptable fallback occurs.

Reply:
I've added some test of unexpected exceptions (wrapped and not wrapped) to verify they are thrown as BridgeServiceExceptions. I have only caught these from update calls, but persistence exceptions could come from other calls. In this case, the exceptions being thrown would differ, so there's an argument to be made that we should just rethrow the exception if we don't want to translate it, rather than wrapping it in BridgeServiceException, which would make the behavior consistent if something like a get() call threw a PersistenceException. BTW I would have told you all of these runtime exceptions were converted by the ExceptionInterceptor to _look_ like BSEs, but it does not. The actually class name is used as the JSON type, etc. though they are all 500 calls.

Reply:
See slack about an approach that would cover all HibernateHelper calls and exception translation